hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -881,6 +881,106 @@ def draw_boolean(
         pass
 
 
+class IRTreeLeaf:
+    def __init__(self, *, ir_type: IRTypeName, value: IRType, kwargs: IRKWargsType):
+        self.ir_type = ir_type
+        self.value = value
+        self.kwargs = kwargs
+
+    def _repr_pretty_(self, p, cycle):
+        assert cycle is False
+        p.text(f"{self.ir_type} {self.value} {self.kwargs}")
```

Comment:
I wrote these pretty printer methods purely for debugging (where they have been enormously helpful). I also wrote one for `DataTree` when I was working on it, but never pushed it. Do you have thoughts in general about pretty-print debug functions and if they should live in the master branch of codebases? We'll have to consider e.g. coverage tests for these, so it's nonzero overhead.

Reply:
I love debugging helpers, and we can just ignore `_repr_pretty_` methods in coverage if that makes it easier - we already do for several others including `__repr__`: https://github.com/HypothesisWorks/hypothesis/blob/3eea8c5ee1ceefea7b97697a102f6d5b5704c892/hypothesis-python/.coveragerc#L19-L22

Reply:
nice! I'm happy with ignoring `_repr_pretty_` for coverage.